### ISSUE TITLE
Proceeding problem

### ISSUE THREAD

**Issue by fredarosus** [miguelfreitas/twister-core#357]
I'm following the instruction page: https://github.com/miguelfreitas/twister-core/blob/master/doc/building-on-ubuntu-debian.md
but get stuck at step Start.
What I encounter is when I proceed the command "./twisterd -rpcuser=user -rpcpassword=pwd -rpcallowip=127.0.0.1" the command cursor flashes and nothing follows up, yet a blank webpage pops up while I click the link:  http://127.0.0.1:28332/index.html.

What's wrong I'm getting stuck with? Any answer is much appreciated.

**Comment by miguelfreitas** [comment]
Please make sure you have loaded twister-html into the following directory:

`~/.twister/html`  (that is `/home/<username>/.twister/html`)

Do you see a file `index.html` there?

**Comment by fredarosus** [comment]
@miguelfreitas Yes, I can see the file index.html there, as a result to which I've used the command the instruction page indicadites:

git clone https://github.com/miguelfreitas/twister-html.git ~/.twister/html

**Comment by miguelfreitas** [comment]
Try watching the log file as you access the page:

`tail -f ~/.twister/debug.log`

For example: If I try to access a non-existing file (`http://127.0.0.1:28332/whatever.html`) I get this:

`ServiceConnection: file /home/miguel/.twister/html/whatever.html not found`

Checking the log should help us understanding what is going wrong.

**Comment by miguelfreitas** [comment]
I've also commited a minimal 404 page to be returned by twisterd.

If you update your twister-core and recompile you shouldn't see a blank page anymore. That was completely unhelpful and annoying!

**Comment by fredarosus** [comment]
@miguelfreitas Thanks to your instruction command 

tail -f ~/.twister/debug.log

Now I can track the instant log infomation rolling over the command line! And I see the most frequently messege stacking up is like:

adding torrent for [xxx, tracker].

While the localhost still shows empty.

Regards.

**Comment by fredarosus** [comment]
I've downloaded the latest source code 0.9.31 and compile them successfully. Then I git-clone the html file into the specific diretory, run the command twisterd and open the localhost, but the problem still remains.

**Comment by Erkan-Yilmaz** [comment]
@fredarosus 

1. those messages in the log are "OK" (I mean: not indicating a problem for you)
2. I assume your twisterd works correct? e.g. you see help message when you execute: ./twisterd help
3. perhaps changing the port in your twister.conf might help? guessing into the blue without much info here

**Comment by fredarosus** [comment]
@Erkan-Yilmaz So kind of you gentleman. Unfortunately, command `./twisterd help` doesn't work for me but rather than an error message returning, it reads `error: server returned HTTP error 403`. Additionlly, my twister.conf file only contains two lines: `rpcuser=user` and `rpcpassword=pwd`. Could you help me one more time?

Best regards.

**Comment by Erkan-Yilmaz** [comment]
1. the port you can change with (here port 40000): rpcport=40000
2. but more important is: did you get any error while compiling your twisterd? I believe if you fix this "error: server returned HTTP error 403", then all will be fine

**Comment by Erkan-Yilmaz** [comment]
@fredarosus

I just looked at (1) which means in short: forbidden
Could it be that your user/pwd is wrong ?
Perhaps, just try to add to the help command from above also your user + pwd.

(1) https://en.wikipedia.org/wiki/HTTP_403

**Comment by miguelfreitas** [comment]
It is kind of strange to see this authentication error since both the running daemon and the client (both twisterd) use the same user/pwd from config file.

Unless one started the daemon with one configuration, change the file, and then tried running `twisterd help`

Also @fredarosus try running latest twister-core from git (not 0.9.31 tarball). Version is the same, but current git version should report a better error message to your browser screen.

**Comment by djmaze** [comment]
Having the same problem using the latest Docker image (which is automatically built from master). Which still worked one or two weeks ago. 

It's strange because nothing significant seems to have been changed in twister-core or twister-html.

Here's the output of curl:

```bash
$ curl -i user:pwd@localhost:28332   
HTTP/1.1 403 Forbidden
Date: Wed, 11 Nov 2015 08:38:08 +0000
Connection: close
Content-Length: 0
Content-Type: application/json
Content-Security-Policy: script-src 'self' 'unsafe-eval'
Server: bitcoin-json-rpc/v0.9.31.0-unk-beta
```

**Comment by fredarosus** [comment]
@Erkan-Yilmaz I'm feeling like so strange that help command returns normally today! But problem's still unsolved. I set my username and password in the twister.conf file, which are `rpcuser=CekJosDaygay`, `rpcpassword=n SD[HY)2BkJ{W.JTng_$?--:`. Is that Okay? I see that incoming error going out as well.

@miguelfreitas Thanks to my poor knowledgement, I can't differ daemon and client. I suppose client includes both twisterd and web interface, am I right? How about daemon?By the way, is there a update command to change the current version to master which I've compiled before made a downgrade? Or only need to recompile it and it will overwrite the old one?

**Comment by miguelfreitas** [comment]
@fredarosus it is a bit confuse indeed: that's because `twisterd` is both daemon (server) and client. When you run it without extra commands (or with `-daemon`) it will sit there waiting to serve local connections. When you pass commands such as `help` it will behave as a client, it will connect to the running server to send commands and receive replies.

What I don't know is if it is legal to have spaces and 's on your `rpcpassword`, they might get stripped out. i'd rather not.

**Comment by miguelfreitas** [comment]
@djmaze are you sure user/pwd is what is actually configured at your `twisterd.conf`? does `twisterd help` works?

**Comment by djmaze** [comment]
@miguelfreitas Okay, found the reason. After the latest Docker update, `rpcallowip` was not correct anymore. #358 fixes the image.
